source/common/grpc/async_client_manager_impl.cc

```diff
@@ -204,7 +204,12 @@ void AsyncClientManagerImpl::RawAsyncClientCache::evictEntriesAndResetEvictionTi
   // Evict all the entries that have expired.
   while (!lru_list_.empty()) {
     MonotonicTime next_expire = lru_list_.back().accessed_time_ + EntryTimeoutInterval;
-    if (now >= next_expire) {
+    if ((now >= next_expire) ||
+        // since 'now' and 'next_expire' are in nanoseconds, the following condition is to check if
+        // the difference between them is less than 1 second. If we don't do this, the timer will
+        // be enabled with 0 seconds, which will cause the timer to fire immediately. This will
+        // cause cpu spike.
+        (std ::chrono::duration_cast<std::chrono::seconds>(next_expire - now).count() == 0)) {
```

Comment:
Do we have a good way to test this? How did you chose 1 second as a good threshold?

Reply:
thanks Lizan for taking a look.
>How did you chose 1 second as a good threshold?
While enabling timer at L218, duration is round-off to seconds. Spike happens when this round-off result is 0. Hence I used same rounding-off in the `if` condition to avoid timer initialiazation with 0s. From the logs, that I shared in PR description as well, when I figured out that if difference between `next_expire` and `now` is less than a sec it gets rounded off to 0, hence mentioned 1 sec in code comment. May be it gets round off to 0 only when diff is less than .5sec, I am not exactly sure about this round-off. 

> Do we have a good way to test this?

I will go through existing tests and see if we have a way to test this out